pyglet/image/__init__.py

```diff
@@ -362,7 +386,10 @@ def get_region(self, x, y, width, height):
         """
         raise ImageException('Cannot get region for %r' % self)
 
-    def save(self, filename=None, file=None, encoder=None):
+    def save(self, filename: Optional[str]=None, 
```

Comment:
Two things: 
`filename` should not default to `None`. This is a bug. Filename is required. 
This function does not return anything, so return type should be `-> None`.